### PR TITLE
Update some ProtocolHandler @spec's

### PR DESCRIPTION
**Pull request by jparise**
start_link/5's @spec wasn't updated when it signature was last changed.
Also add a @spec to init/6 and mark some internal functions as private.

### CODE REVIEW

File: lib/thrift/binary/framed/protocol_handler.ex [hunk 1]
```diff
@@ -11,12 +11,13 @@ defmodule Thrift.Binary.Framed.ProtocolHandler do
   }
   require Logger
 
-  @spec start_link(reference, pid, module, {module, module}) :: GenServer.on_start
+  @spec start_link(reference, pid, module, {module, module, Keyword.t}) :: GenServer.on_start
```
Comment (dantswain) [inline lib/thrift/binary/framed/protocol_handler.ex]: Is it worth defining a type for tcp_opts?  Can be helpful for documentation purposes.

Comment (dantswain) [inline lib/thrift/binary/framed/protocol_handler.ex]: Should we tighten down what `transport` is allowed to be?  Not just in the spec but also in `start_link` - `when transport in [OptionA, OptionB]`?

Comment (jparise) [inline lib/thrift/binary/framed/protocol_handler.ex]: Have a look at what I just pushed. I think it's a reasonable balance between the semi-generic nature of the ProtocolHandler and the fact that it's implementation is `:ranch_tcp`-specific right now.

File: lib/thrift/binary/framed/protocol_handler.ex [hunk 2]
```diff
@@ -5,20 +5,26 @@ defmodule Thrift.Binary.Framed.ProtocolHandler do
...
-    pid = spawn_link(__MODULE__, :init, [ref, socket, transport, server_module, handler_module, tcp_opts])
+  @spec start_link(reference, port, transport, {module, module, transport_opts}) :: GenServer.on_start
+  def start_link(ref, socket, transport, {server_module, handler_module, transport_opts}) do
+    pid = spawn_link(__MODULE__, :init, [ref, socket, transport, server_module, handler_module, transport_opts])
     {:ok, pid}
   end
 
-  @spec init(reference, pid, module, module, module, Keyword.t) :: :ok | no_return
-  def init(ref, socket, transport, server_module, handler_module, tcp_opts) do
+  @spec init(reference, port, :ranch_tcp, module, module, :ranch_tcp.opts) :: :ok | no_return
```
Comment (dantswain) [inline lib/thrift/binary/framed/protocol_handler.ex]: What happens when this is `:ranch_ssl`?

Comment (jparise) [inline lib/thrift/binary/framed/protocol_handler.ex]: `init/6` doesn't support `:ranch_ssl`; I added the `:ranch_tcp` pattern match the function signature to lock that in.

We should need to write a new `:ranch_ssl`-specific implementation to support it.

I guess, given that, I should drop `:ranch_ssl` from the `transport` type for now?

Comment (dantswain) [inline lib/thrift/binary/framed/protocol_handler.ex]: Ah I didn't realize that.

> I guess, given that, I should drop :ranch_ssl from the transport type for now?

Yeah that makes sense to me.  Maybe an issue so someone remembers to go back and support it?

Comment (jparise) [inline lib/thrift/binary/framed/protocol_handler.ex]: Filed as #214

**Pull request merged**